### DISCUSSION TITLE
Does eatch iteration recomputing whole definition?

### DISCUSSION THREAD

**Discussion by Fariultra** [Tunny-gh/Tunny_in_Open_Sea#260] [General]
Hi, I was testing a few days ago tunny and experiencing significantly slower iterations compared to the Galapagos. My definition is quite demanding due to geometrical operations. To make it faster I preprocessed a couple of operations before being affected by gene pool variables. I got the impression that Tunny recomputing a whole definition every iteration, is it the case?

**Comment by hrntsm** [comment]
Hi @Fariultra,

Thank you for your interest.
I am aware that the problem is that the calculations are slower than Galapagos because everything is recalculated as you mentioned.
Do you know of a better solution?

**Reply by Fariultra** [reply]
Hi @hrntsm,

 I forgot to thank you for your work in the first place. I assume that the recalculation of only components after variables is not simply doable? Somehow I am unable to explain so tried to draw:
 
![image](https://github.com/hrntsm/Tunny/assets/39439308/230f3306-69df-498b-9ea8-49bdb8d38430)

The current state is the RED lines?

**Reply by hrntsm** [reply]
Hi @Fariultra, Thanks your drawing.

as you mentioned current is red line.
I will try to change tunny operation to yellow and blue line.

**Reply by hrntsm** [reply]
Hi @Fariultra 
I fixed tunny to your blue line.

Could you test below version?

https://github.com/hrntsm/Tunny/actions/runs/8323613670/artifacts/1334399744

**Reply by hrntsm** [reply]
Fix this in v0.12
